OsmAnd/src/net/osmand/plus/views/layers/POIMapLayer.java

```diff
@@ -602,26 +596,29 @@ public LatLon getObjectLocation(Object o) {
 	public boolean showMenuAction(@Nullable Object object) {
 		OsmandApplication app = view.getApplication();
 		MapActivity mapActivity = view.getMapActivity();
-		TravelHelper travelHelper = app.getTravelHelper();
 		if (mapActivity != null && object instanceof Amenity) {
+			TravelHelper travelHelper = app.getTravelHelper();
 			Amenity amenity = (Amenity) object;
-			if (amenity.getSubType().equals(ROUTE_TRACK)) {
-				TravelGpx travelGpx = travelHelper.searchGpx(amenity.getLocation(), amenity.getRouteId(), amenity.getRef());
-				if (travelGpx == null) {
+			String subType = amenity.getSubType();
+			if (amenity.getType().getKeyName().equals(ROUTES)) {
+				if (subType.equals(ROUTE_ARTICLE)) {
+					String lang = app.getLanguage();
+					lang = amenity.getContentLanguage(Amenity.DESCRIPTION, lang, "en");
+					String name = amenity.getGpxFileName(lang);
+					TravelArticle article = travelHelper.getArticleByTitle(name, lang, true, null);
+					if (article == null) {
+						return true;
+					}
+					travelHelper.openTrackMenu(article, mapActivity, name, amenity.getLocation(), false);
 					return true;
-				}
-				travelHelper.openTrackMenu(travelGpx, mapActivity, amenity.getRouteId(), amenity.getLocation());
-				return true;
-			} else if (amenity.getSubType().equals(ROUTE_ARTICLE)) {
-				String lang = app.getLanguage();
-				lang = amenity.getContentLanguage(Amenity.DESCRIPTION, lang, "en");
-				String name = amenity.getName(lang);
-				TravelArticle article = travelHelper.getArticleByTitle(name, lang, true, null);
-				if (article == null) {
+				} else if (amenity.isRouteTrack()) {
+					TravelGpx travelGpx = travelHelper.searchGpx(amenity.getLocation(), amenity.getRouteId(), amenity.getRef());
+					if (travelGpx == null) {
+						return true;
+					}
+					travelHelper.openTrackMenu(travelGpx, mapActivity, amenity.getGpxFileName(null), amenity.getLocation(), false);
 					return true;
```

Comment:
looks like return true; in npe check is unnecessary, just move openTrackMenu in this check

Reply:
a81a228f7f